### PR TITLE
Bucket for Program Cohort status

### PR DESCRIPTION
**Pull request by bmuniswamy**
# Critical Changes

# Changes
Created a new bucket for Program Cohort Status field. Now Admins can group Program Cohort Statuses under this bucket so they can filter out the Cohorts that display in the picklist that is displayed on the Participant Selector screen in the BSDT group wizard and New Service Schedule Wizard.
# Issues Closed
[W-10999467](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000uxqTHYAY/view)
# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~- [ ] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage~~  
~~- [ ] CRUD/FLS is enforced in Apex~~
~~- [ ] Permission sets are updated to account for CRUD|FLS|Tab|Classes~~
~~- [ ] Field sets are updated to account for new fields~~
~~- [ ] UX approval or UX not necessary~~
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [x] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

File: force-app/main/default/classes/ProgramSelector.cls [hunk 1]
```diff
@@ -16,11 +22,33 @@ public with sharing class ProgramSelector {
...
-            WHERE Program__c = :programId
+            WHERE Program__c = :programId AND Status__c IN :allowedProgramCohortStatuses
             ORDER BY Name ASC
         ];
 
         return Security.stripInaccessible(AccessType.READABLE, programCohorts)
             .getRecords();
     }
+
+    public Set<String> getAllowedProgramCohortStatusValues() {
```
Comment (jjbennett) [inline force-app/main/default/classes/ProgramSelector.cls:33]: Should this be private? And do we need to add a test case to validate getProgramCohortsFromProgramId now only returns active? Do we want to consider modifying the method name to getActiveProgramCohorts?

Comment (jjbennett) [inline force-app/main/default/classes/ProgramSelector.cls:33]: Should this be private? And do we need to add a test case to validate getProgramCohortsFromProgramId now only returns active? Do we want to consider modifying the method name to getActiveProgramCohorts?

File: force-app/main/default/classes/ProgramSelector_TEST.cls [hunk 2]
```diff
@@ -40,6 +42,68 @@ public with sharing class ProgramSelector_TEST {
...
+        programSelector.bucketSelector = (FieldBucketSelector) bucketSelectorStub.create();
+
+        Test.startTest();
+        actualCohorts = programSelector.getProgramCohortsFromProgramId(programId);
+        Test.stopTest();
+
+        System.assertEquals(
+            1,
+            actualCohorts.size(),
+            'Actual results not the same as expected results'
```
Comment (jjbennett) [inline force-app/main/default/classes/ProgramSelector_TEST.cls:89]: Can we be more specific in these messages for example Expected one active cohort to be returned? Also if we know there is only one do we need to loop it or just analyze that one record?

File: force-app/main/default/classes/ProgramSelector_TEST.cls [hunk 3]
```diff
@@ -40,6 +42,68 @@ public with sharing class ProgramSelector_TEST {
...
+        List<String> buckets = new List<String>{ ACTIVE_BUCKET };
+        Schema.SObjectType programCohortSObjType = ProgramCohort__c.SObjectType;
+        Schema.SObjectField statusField = ProgramCohort__c.Status__c;
+
+        TestDataFactory.generateProgramData();
+        Id programId = TestDataFactory.programs[0].Id;
+
+        List<Bucket__mdt> cohortStatusBuckets = createBuckets();
+
+        //Insert a cohort with different status so we check that we return
```
Comment (jjbennett) [inline force-app/main/default/classes/ProgramSelector_TEST.cls:57]: Should we add something like this to the previous test to ensure we are getting all records back if there is no bucket?

File: force-app/main/default/classes/ProgramSelector_TEST.cls [hunk 4]
```diff
@@ -69,4 +133,35 @@ public with sharing class ProgramSelector_TEST {
...
         );
     }
+
+    private static List<Bucket__mdt> createBuckets() {
+        BucketedField__mdt bucketedField = (BucketedField__mdt) new TestUtil.BucketedFieldBuilder()
+            .withDeveloperName('ProgramCohortStatuses')
+            .withQualifiedApiName(Util.prefixNamespace('ProgramCohortStatuses'))
+            .withField('Status__c')
+            .withObject('ProgramCohort__c')
+            .withMockId()
```
Comment (jjbennett) [inline force-app/main/default/classes/ProgramSelector_TEST.cls:143]: If we aren't setting a mockId do we need to call this method?

Comment (bmuniswamy) [reply force-app/main/default/classes/ProgramSelector_TEST.cls:143]: Sorry I missed it. I will go ahead and remove it. Thanks for catching it.

File: force-app/main/default/classes/ProgramSelector_TEST.cls (hunk 2 above)
Comment (bmuniswamy) [reply force-app/main/default/classes/ProgramSelector_TEST.cls:89]: I was debating it and then decided to add the loop. I will remove the loop

**Review by jjbennett — APPROVED** [review]
Works Great! We should probably mention in the release notes that the admin will need to create a bucket value in order for the filtering to work.

**Pull request merged**